app/models/merchant.rb

```diff
@@ -14,6 +14,13 @@ def revenue(date)
       .sum('unit_price_in_cents * quantity')
   end
 
+  def self.most_revenue(number_of_merchants)
+    joins(:invoice_items)
+      .merge(InvoiceItem.successful)
```

Comment:
I like where you are headed with this, but this is the same as `.merge(Transaction.successful)` so I would prefer using Transaction instead of going through InvoiceItem first.